tls/test_alpn_order.py

```diff
@@ -0,0 +1,63 @@
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2019 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+import ssl
+import socket
+from copy import deepcopy
+
+from framework import tester
+from helpers import tf_cfg
+
+
+class TestALPNOrderBase(tester.TempestaTest, base=True):
+    tempesta_template = {
+        "config": """
+        listen 443 proto=%(tempesta_proto)s;
+
+        tls_match_any_server_name;
+
+        srv_group default {
+            server ${server_ip}:8000;
+        }
+
+        vhost tempesta-tech.com {
+           tls_certificate ${tempesta_workdir}/tempesta.crt;
+           tls_certificate_key ${tempesta_workdir}/tempesta.key;
+           proxy_pass default;
+        }
+        """
+    }
+
+    def setUp(self):
+        super(TestALPNOrderBase, self).setUp()
+        self.tempesta = deepcopy(self.tempesta_template)
+        self.tempesta["config"] = self.tempesta["config"] % {
+            "tempesta_proto": self.tempesta_proto,
+        }
+        self.start_tempesta()
+        self.assertTrue(self.wait_all_connections())
+
+    def test(self):
+        hostname = tf_cfg.cfg.get("Tempesta", "hostname")
+        context = ssl.create_default_context()
+        self.assertTrue(ssl.HAS_ALPN, "`ssl` library backend does not support ALPN")
+        context.set_alpn_protocols(self.protocols)
+
+        with socket.create_connection((hostname, 443)) as sock:
+            with context.wrap_socket(sock, server_hostname=hostname) as ssock:
+                self.assertEqual(
+                    ssock.selected_alpn_protocol(),
+                    self.protocols[0] if len(self.protocols) else "",
+                    "wrong protocol has been prioritized",
+                )
+
+
+class TestALPNOrderH1(TestALPNOrderBase):
+    tempesta_proto = "https"
+    protocols = ["http/1.1", "http/2"]
+
+
+class TestALPNOrderH2(TestALPNOrderBase):
+    tempesta_proto = "h2"
+    protocols = ["http/2", "http/1.1"]
```

Comment:
Could you please try my branch from #1643? At least yesterday I successfully played with ALPNs https://github.com/tempesta-tech/tempesta-test/pull/275#discussion_r948549593 . Also in my understanding, ALPNs are `https` and `h2`, not `http/2` and `http/1.1`.

Reply:
Currently that branch fails to build with:
```
ERROR: modpost: "tcp_write_queue_purge" [/root/tempesta/fw/tempesta_fw.ko] undefined!
```
Regarding protocol names, you're probably right, I just took those from the docs https://docs.python.org/3/library/ssl.html#ssl.SSLContext.set_alpn_protocols

Reply:
Yeah, the build issue is the subject for #79